src/main/java/io/cdap/plugin/zuora/restobjects/objects/BaseResult.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright © 2019 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package io.cdap.plugin.zuora.restobjects.objects;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.annotations.SerializedName;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * API Response Wrapper
+ *
+ * @param <T> Any {@link BaseObject} object
+ */
+public class BaseResult<T> {
+  @SerializedName("result")
+  private List<T> result;
+
+  @SerializedName(value = "nextPage")
+  private String nextPage;
+
+  @SerializedName(value = "success")
+  private Boolean success;
+
+  @SerializedName(value = "cdapObjectName")
+  private String cdapObjectName;
+
+  public List<T> getResult() {
+    return result;
+  }
+
+  public Map<String, String> getNextPageArguments() {
+    if (Strings.isNullOrEmpty(nextPage)) {
+      return null;
+    }
+    URL url;
+    try {
+      url = new URL("http://" + nextPage);
+    } catch (MalformedURLException e) {
+      return null;
+    }
+    ImmutableMap.Builder<String, String> builder = new ImmutableMap.Builder<>();
+    Arrays.stream(url.getQuery().split("&"))
+      .map(x -> x.split("="))
+      .forEach(x -> builder.put(x[0], x[1]));
+    return builder.build();
+  }
+
+  public String getNextPage() {
+    if (Strings.isNullOrEmpty(nextPage)) {
+      return null;
+    }
+
+    URL url;
+    try {
+      url = new URL("http://" + nextPage);
+    } catch (MalformedURLException e) {
+      return null;
```

Comment:
The default value for this property is null, which is returned by API. It means that no new page provided and server response is completed at last page. 

If the provided url by the server is malformed, it could be caused by an exception on back-end and should not ddos api with wrong api and non-existing page. (i.e. we considering wrong url as the last result page in the query)

Reply:
added comment to the code